commit 2ba459685204af53b034d269d5cdb3059d4b471e
Author: David Hildenbrand <dahi@linux.vnet.ibm.com>
Date:   Wed Mar 25 13:12:32 2015 +0100

    KVM: s390: store the breaking-event address on pgm interrupts
    
    If the PER-3 facility is installed, the breaking-event address is to be
    stored in the low core.
    
    There is no facility bit for PER-3 in stfl(e) and Linux always uses the
    value at address 272 no matter if PER-3 is available or not.
    We can't hide its existence from the guest. All program interrupts
    injected via the SIE automatically store this information if the PER-3
    facility is available in the hypervisor. Also the itdb contains the
    address automatically.
    
    As there is no switch to turn this mechanism off, let's simply make it
    consistent and also store the breaking event address in case of manual
    program interrupt injection.
    
    Reviewed-by: Jens Freimann <jfrei@linux.vnet.ibm.com>
    Signed-off-by: David Hildenbrand <dahi@linux.vnet.ibm.com>
    Reviewed-by: Christian Borntraeger <borntraeger@de.ibm.com>
    Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>
    Acked-by: Cornelia Huck <cornelia.huck@de.ibm.com>

diff --git a/arch/s390/kvm/interrupt.c b/arch/s390/kvm/interrupt.c
index 2afec6006def..2361b8ed0a50 100644
--- a/arch/s390/kvm/interrupt.c
+++ b/arch/s390/kvm/interrupt.c
@@ -585,6 +585,8 @@ static int __must_check __deliver_prog(struct kvm_vcpu *vcpu)
 		kvm_s390_rewind_psw(vcpu, ilc);
 
 	rc |= put_guest_lc(vcpu, ilc, (u16 *) __LC_PGM_ILC);
+	rc |= put_guest_lc(vcpu, vcpu->arch.sie_block->gbea,
+				 (u64 *) __LC_LAST_BREAK);
 	rc |= put_guest_lc(vcpu, pgm_info.code,
 			   (u16 *)__LC_PGM_INT_CODE);
 	rc |= write_guest_lc(vcpu, __LC_PGM_OLD_PSW,